lib/reek/context/code_context.rb

```diff
@@ -16,13 +15,16 @@ module Context
     #
     # :reek:TooManyMethods: { max_methods: 19 }
     # :reek:TooManyInstanceVariables: { max_instance_variables: 8 }
+    # :reek:Attribute
     class CodeContext
       extend Forwardable
       delegate each_node: :exp
       delegate %i(name type) => :exp
-      delegate %i(visibility visibility= non_public_visibility?) => :visibility_tracker
 
-      attr_reader :children, :context, :exp, :statement_counter, :visibility_tracker
+      attr_accessor :visibility
```

Comment:
I like the separation of the property 'visibility', and the thing that tracks the visibility. In fact, only methods and attributes have a visibility, while only modules and classes need a tracker. Ideally, I would therefore push down both the visibility tracker and visibility to the contexts that actually need them. That should clear up the confusion as well.